tools/cmd/api-generator/convert_metadata.go

```diff
@@ -42,54 +45,73 @@ func specToMetadata(spec *openapi3.T) (map[string]*metadatatypes.Metadata, error
 }
 
 // Returns a map of operationID:*Metadata.
-func extractMetadata(operation *openapi3.Operation) (*metadatatypes.Metadata, error) {
+func extractMetadata(operation *openapi3.Operation) (*metadatatypes.OperationMetadata, error) {
 	if operation == nil {
 		return nil, nil
 	}
 
+	paramMetadata := extractParameterMetadata(operation.Parameters)
+
 	requestBodyExamples, err := extractRequestBodyExamples(operation.RequestBody)
 	if err != nil {
 		return nil, err
 	}
 
-	paramMap := extractParameterMetadata(operation.Parameters)
+	paramExamples := extractParameterExamples(operation.Parameters)
 
-	return &metadatatypes.Metadata{
-		Parameters:          paramMap,
-		RequestBodyExamples: requestBodyExamples,
+	examples, err := buildExamples(requestBodyExamples, paramExamples, operation)
+	if err != nil {
+		return nil, err
+	}
+
+	return &metadatatypes.OperationMetadata{
+		Parameters: paramMetadata,
+		Examples:   examples,
 	}, nil
 }
 
-// For each parameter in an operation, the parameter name and example is extracted.
-// A map of parameterName:example is returned.
 func extractParameterMetadata(parameters openapi3.Parameters) map[string]metadatatypes.ParameterMetadata {
 	result := make(map[string]metadatatypes.ParameterMetadata)
 
 	for _, parameterRef := range parameters {
-		example := ""
-		if parameterExample, ok := parameterRef.Value.Schema.Value.Example.(string); ok {
-			example = parameterExample
-		}
 		result[parameterRef.Value.Name] = metadatatypes.ParameterMetadata{
-			Example: example,
-			Usage:   parameterRef.Value.Description,
+			Usage: parameterRef.Value.Description,
+		}
+	}
+
+	return result
+}
+
+type extractedExamples struct {
+	Example  any
+	Examples openapi3.Examples
+}
+
+func extractParameterExamples(parameters openapi3.Parameters) map[string]extractedExamples {
+	result := make(map[string]extractedExamples)
+
+	for _, parameterRef := range parameters {
+		defaultExample := parameterRef.Value.Example
+		if defaultExample == nil {
+			defaultExample = parameterRef.Value.Schema.Value.Example
```

Comment:
[q] Do you think we should add some null checks here, or are you certain that none of these fields will ever be nil?